packages/visx-xychart/src/components/Tooltip.tsx

```diff
@@ -7,18 +7,54 @@ import { UseTooltipPortalOptions } from '@visx/tooltip/lib/hooks/useTooltipInPor
 import TooltipContext from '../context/TooltipContext';
 import DataContext from '../context/DataContext';
 import { TooltipContextType } from '../types';
+import getScaleBandwidth from '../utils/getScaleBandwidth';
 
-export type RenderTooltipParams = TooltipContextType & {
+/** fontSize + lineHeight from default styles break precise location of crosshair, etc. */
+const TOOLTIP_NO_STYLE: React.CSSProperties = {
+  position: 'absolute',
+  pointerEvents: 'none',
+  fontSize: 0,
+  lineHeight: 0,
+};
+
+type CircleProps = {
+  left?: number;
+  top?: number;
+  fill?: string;
+  stroke?: string;
+  strokeWidth: number;
+  radius: number;
+};
+
+export type RenderTooltipParams<Datum extends object> = TooltipContextType<Datum> & {
   colorScale?: PickD3Scale<'ordinal', string, string>;
 };
 
-export type TooltipProps = {
+export type TooltipProps<Datum extends object> = {
   /**
    * When TooltipContext.tooltipOpen=true, this function is invoked and if the
    * return value is non-null, its content is rendered inside the tooltip container.
    * Content will be rendered in an HTML parent.
    */
-  renderTooltip: (params: RenderTooltipParams) => React.ReactNode;
+  renderTooltip: (params: RenderTooltipParams<Datum>) => React.ReactNode;
+  /** Whether to snap tooltip + crosshair x-coord to the nearest Datum x-coord instead of the event x-coord. */
+  snapTooltipToDatumX?: boolean;
+  /** Whether to snap tooltip + crosshair y-coord to the nearest Datum y-coord instead of the event y-coord. */
+  snapTooltipToDatumY?: boolean;
+  /** Whether to show a vertical line at tooltip position. */
+  showVerticalCrosshair?: boolean;
+  /** Whether to show a horizontal line at tooltip position. */
+  showHorizontalCrosshair?: boolean;
+  /** Whether to show a circle at the tooltip position for the (single) nearest Datum. */
+  showDatumCircle?: boolean;
```

Comment:
Perhaps name this `showDatumMark` or `showDatumMarker` ? In case you want to support other shapes (that is not circle) in the future.

Reply:
good call. maybe I should make these use `@visx/glyph` 🤔  maybe I'll do that as a followup PR.

Reply:
updated to `glyph` throughout with the future glyph support in mind